src/app/mesh/qgsmaptooleditmeshframe.h

```diff
@@ -0,0 +1,247 @@
+      AddingNewFace, //!< Adding a face has been start and the user have to choose or digitize vertices
+      Selecting, //!< Selection is in process
+      MovingVertex //!< Moving vertex or vertices is processing
+    };
+
+    typedef std::array<int, 2> Edge; //first face index, second the vertex index corresponding to the end extremity (ccw)
+
+    // methods
+    void initialize();
+    void activateWithState( State state );
+    const QgsMeshVertex mapVertex( int index ) const;
+    const QgsPointXY mapVertexXY( int index ) const;
+    const QgsMeshFace nativeFace( int index ) const;
+
+    void highLight( const QgsPointXY &mapPoint );
+    void highlightCurrentHoveredFace( const QgsPointXY &mapPoint );
+    void highlightCloseVertex( const QgsPointXY &mapPoint );
+    void highlightCloseEdge( const QgsPointXY &mapPoint );
+    bool edgeCanBeInteractive( int vertexIndex1, int vertexIndex2 ) const;
+    bool faceCanBeInteractive( int faceIndex ) const;
+
+    void createZValueWidget();
+    void deleteZvalueWidget();
+
+    void clearSelection();
+    void clearCanvasHelpers();
+    void clearEdgeHelpers();
+
+    void clearAll();
+
+    void addVertex( const QgsPointXY &mapPoint, const QgsPointLocator::Match &mapPointMatch );
+    void updateFreeVertices();
+
+    //! Checks if we are closed to a vertex, if yes return the index of the vertex;
+    int closeVertex( const QgsPointXY &point ) const;
+
+    QgsPointSequence nativeFaceGeometry( int faceIndex ) const;
+    QVector<QgsPointXY> edgeGeometry( const Edge &edge ) const;
+    QVector<int> edgeVertices( const Edge &edge ) const;
+
+    QgsPointXY newFaceMarkerPosition( int vertexIndex );
+
+    void addVertexToFaceCanditate( int vertexIndex );
+    bool testNewVertexInFaceCanditate( int vertexIndex );
+
+    // selection methods
+    void select( const QgsPointXY &mapPoint, Qt::KeyboardModifiers modifiers, double tolerance );
+    void setSelectedVertices( const QList<int> newSelectedVertex,  Qt::KeyboardModifiers modifiers );
+    void clearSelectedvertex();
+    void selectInGeometry( const QgsGeometry &geometry,  Qt::KeyboardModifiers modifiers );
+    void applyZValueOnSelectedVertices();
+    void prepareSelection();
+    void updateSelectecVerticesMarker();
+
+    bool testBorderMovingFace( const QgsMeshFace &borderMovingfaces, const QgsVector &translation ) const;
+
+    // members
+    struct SelectedVertexData
+    {
+      //Here edges are the indexes of the face where the following vertices (ccw) is the other extremity of the edge
+      QList<Edge> selectedEdges;
+      QList<Edge> meshFixedEdges;
+    };
+
+    bool mIsInitialized = false;
+    State mCurrentState = Digitizing;
+    State mPreviousState = Digitizing; //used to store a state and restore it after a particular action as selecting
+    bool mLeftButtonPressed = false;
+
+    QPointer<QgsMeshLayer> mCurrentLayer = nullptr;
```

Comment:
can you add to all these pointers comment if it is owned by the class or someone else?